### PR TITLE
add info that retaining resources is not supported in sandbox

### PR DESCRIPTION
**Pull request by rtpascual**
#### Description of changes:

Add callout that retaining resources on stack deletion is not supported for sandbox.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by josefaidt — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx [hunk 1]
```diff
@@ -164,6 +164,12 @@ plan.addSelection("BackupPlanSelection", {
 
 For example, if you would like to retain a resource on stack deletion, you can use the `applyRemovalPolicy` property on the resource to add a retention policy.
 
+<Callout>
+
+Retaining resources on stack deletion is not supported in sandbox.
```
Comment (Amplifiyer) [inline src/pages/[platform]/build-a-backend/add-aws-services/deletion-backup-resources/index.mdx:169]: I'd reword this to something like
"`sandbox delete` ignores any resource removal policy and deletes all resources always."

Since the stack deletion outside of amplify would still respect these removal policy (e.g. using CDK or CFN directly)

**Review by josefaidt — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]
